apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -250,26 +289,18 @@ public static void ReleaseToPool(this SpannerClient client, Session session)
                 SessionPoolImpl targetPool = s_poolByClientAndDatabase.GetOrAdd(poolKey,
                                              key => new SessionPoolImpl(key));
 
-
-                SpannerClient evictionClient = poolKey.Client;
-                Session evictionSession = null;
-
                 //Figure out if we want to pool this released session.
-                lock (s_priorityListSync)
+                targetPool.ReleaseSessionToPool(client, session);
+                s_priorityList.Add(targetPool);
+                if (CurrentPooledSessions > MaximumPooledSessions)
```

Comment:
This only evicts a single element, whereas elsewhere we have a while loop. Is that okay?